### PR TITLE
Test cleanup issue 1287

### PR DESCRIPTION
**Pull request by sudarshan-parvatikar**
I have refactored the code according to Issue:1287. I also re-ordered import statements to make them consistent across all files and ran tests. The tests passed and I thinks this should close Issue:1287.

### CODE REVIEW

**Comment by drvinceknight** [comment]
There seems to be a merge conflict, can you rebase this on to master?

File: axelrod/tests/strategies/test_cycler.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
+import unittest
+
 import itertools
+
 import random
-import unittest
 
-import axelrod
+import axelrod as axl
 from axelrod import AntiCycler, Cycler, EvolvableCycler
```
Comment (gaffney2010) [inline axelrod/tests/strategies/test_cycler.py:10]: Delete this line and update references throughout file.

File: axelrod/tests/strategies/test_evolvable_player.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
+import unittest
+
 import functools
+
 import random
-import unittest
 
 import axelrod as axl
 from axelrod import EvolvablePlayer, seed
```
Comment (gaffney2010) [inline axelrod/tests/strategies/test_evolvable_player.py:8]: Delete this line, and update.

File: axelrod/tests/strategies/test_memoryone.py [hunk 3]
```diff
@@ -1,24 +1,25 @@
 """Tests for the Memoryone strategies."""
 
 import unittest
+
 import warnings
 
-import axelrod
+import axelrod as axl
 from axelrod import Game
```
Comment (gaffney2010) [inline axelrod/tests/strategies/test_memoryone.py:8]: Delete line and update

File: axelrod/tests/strategies/test_player.py [hunk 4]
```diff
@@ -1,22 +1,27 @@
+import unittest
```
Comment (gaffney2010) [inline axelrod/tests/strategies/test_player.py:1]: Should there be space between each of these imports?

File: axelrod/tests/strategies/test_player.py [hunk 5]
```diff
@@ -1,22 +1,27 @@
...
 import random
+
 import types
-import unittest
 
-import axelrod
 import numpy as np
+
+import axelrod as axl
 from axelrod import DefaultGame, Player, LimitedHistory
```
Comment (gaffney2010) [inline axelrod/tests/strategies/test_player.py:14]: Delete and update

File: axelrod/tests/unit/test_deterministic_cache.py [hunk 6]
```diff
@@ -1,16 +1,18 @@
+import unittest
+
 import os
+
 import pickle
```
Comment (gaffney2010) [inline axelrod/tests/unit/test_deterministic_cache.py:5]: Spacing here too.

File: axelrod/tests/unit/test_moran.py [hunk 7]
```diff
@@ -1,24 +1,28 @@
+import unittest
```
Comment (gaffney2010) [inline axelrod/tests/unit/test_moran.py:1]: Spacing again

File: axelrod/tests/unit/test_moran.py [hunk 8]
```diff
@@ -1,24 +1,28 @@
...
 import random
-import unittest
+
 from collections import Counter
 
-import axelrod
 import matplotlib.pyplot as plt
+
+import axelrod as axl
 from axelrod import ApproximateMoranProcess, MoranProcess, Pdf
```
Comment (gaffney2010) [inline axelrod/tests/unit/test_moran.py:12]: Delete and update

File: axelrod/tests/unit/test_resultset.py [hunk 9]
```diff
@@ -1,18 +1,23 @@
...
-import axelrod
-import axelrod.interaction_utils as iu
 import pandas as pd
+
+from dask.dataframe.core import DataFrame
+
+from numpy import mean, nanmedian, std
+
+import axelrod as axl
+import axelrod.interaction_utils as iu
```
Comment (gaffney2010) [inline axelrod/tests/unit/test_resultset.py:14]: Looking at the issue.  It looks like it was decide that only axelrod would be imported like this, and that you'd use axl.interaction_utils elsewhere.

File: axelrod/tests/unit/test_tournament.py [hunk 10]
```diff
@@ -1,19 +1,29 @@
 """Tests for the main tournament class."""
 
+import unittest
```
Comment (gaffney2010) [inline axelrod/tests/unit/test_tournament.py:3]: spacing

File: axelrod/tests/unit/test_interaction_utils.py [hunk 11]
```diff
@@ -1,12 +1,13 @@
-import tempfile
 import unittest
+
+import tempfile
+
 from collections import Counter
 
-import axelrod
+import axelrod as axl
 import axelrod.interaction_utils as iu
```
Comment (gaffney2010) [inline axelrod/tests/unit/test_interaction_utils.py:8]: Delete this line.  Ditto from comment on other file.

File: axelrod/tests/unit/test_pickling.py [hunk 12]
```diff
@@ -1,6 +1,8 @@
+import unittest
+
 import pickle
+
 import random
-import unittest
 
 import axelrod as axl
 import axelrod.strategy_transformers as st
```
Comment (gaffney2010) [inline axelrod/tests/unit/test_pickling.py:8]: Delete this line.  It looks like that is what was decided on another file.

**Comment by sudarshan-parvatikar** [comment]
> This is a really nice clean up.
> 
> It looks like there is a few lines missed.
> 
> It also looks like your editor changed the import order of unrelated imports. I don't think we've been consistent about import order. But some of these changes add a lot of new lines, and I think we probably don't want to do that.

Hey @gaffney2010! . Thank you for your helpful comments. I reordered the import statements so that they'd be consistent. However, you're right about adding new lines, so I'll update them as per your reviews. Furthermore, is it ok if the import statements be as in [PEP 8 import rules](https://www.python.org/dev/peps/pep-0008/#imports), as explained by this [answer on SO](https://stackoverflow.com/a/47448739). This may possibly make the import statements more consistent in the future.

**Comment by gaffney2010** [comment]
> Furthermore, is it ok if the import statements be as in PEP 8 import rules, as explained by this answer on SO.

Yes.

File: axelrod/tests/unit/test_resultset.py (hunk 9 above)
Comment (sudarshan-parvatikar) [reply axelrod/tests/unit/test_resultset.py:14]: Yeah, that's right.

File: axelrod/tests/unit/test_interaction_utils.py (hunk 11 above)
Comment (sudarshan-parvatikar) [reply axelrod/tests/unit/test_interaction_utils.py:8]: > Ditto from comment on other file.

Can you quote the comment that you're talking about?

**Comment by sudarshan-parvatikar** [comment]
I have updated the code as per the recent review. Imports are grouped according to the PEP-8 convention, and new lines between the import statements are removed

File: axelrod/tests/unit/test_interaction_utils.py (hunk 11 above)
Comment (gaffney2010) [reply axelrod/tests/unit/test_interaction_utils.py:6]: import axelrod.interaction_utils as iu

It looks like it was decided to just import axelrod, and to replace iu with axelrod.interaction_utils.

**Comment by gaffney2010** [comment]
This looks good aside from the three places where you have:
import axelrod.xxx as yyy

I think it was decided in the issue to only import axelrod.

**Comment by sudarshan-parvatikar** [comment]
> This looks good aside from the three places where you have:
> import axelrod.xxx as yyy
> 
> I think it was decided in the issue to only import axelrod.

Yes, I now understand what you meant to say. I'll fix it asap :)

**Comment by drvinceknight** [comment]
Thanks @sudarshan-parvatikar all looks good to me :+1:

**Comment by sudarshan-parvatikar** [comment]
Thanks all for being so patient and helpful. 😄

**Pull request merged**